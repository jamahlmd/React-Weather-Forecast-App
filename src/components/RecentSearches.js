import React from 'react';
import {Button} from 'antd';
import propTypes from 'prop-types';

//Map over the recentSearches array and display every recent search as a Button
//If the recent search is the same as the current city give it the 'primary' class otherwise the 'default' class
const RecentSearches = ({recent,current,onClick,loading}) => (
    <div
        className="recent"
        style={{paddingTop: '8px'}}>
        {
            recent.map( (rec) => <Button
                loading={loading}
                onClick={() => onClick(rec)}
                style={{marginRight: '8px',textTransform: 'capitalize'}}
                key={rec}
                type={rec === current ? 'primary' : 'default'}
                size={'small'}
            >{rec}</Button>)
        }
    </div>
);

RecentSearches.propTypes = {
  recent: propTypes.array.isRequired,
  current: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
  loading: propTypes.bool.isRequired,
};

export default RecentSearches;