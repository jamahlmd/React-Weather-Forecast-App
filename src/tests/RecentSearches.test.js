import React from 'react';
import {Button} from 'antd';
import RecentSearches from '../components/RecentSearches';

const recent = ['London','Prague'];

const current = "London";

const wrapper = shallow(<RecentSearches recent={recent} current={current} onClick={()=>true} loading={false} />);


test('Should render correctly', () => {

    expect(toJson(wrapper)).toMatchSnapshot();
});

test('Should highlight current search', () => {

   expect(wrapper.find(Button).first().prop('type')).toEqual('primary');
});