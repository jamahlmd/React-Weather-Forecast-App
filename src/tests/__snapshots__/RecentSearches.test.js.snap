// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="recent"
  style={
    Object {
      "paddingTop": "8px",
    }
  }
>
  <Button
    block={false}
    ghost={false}
    key="London"
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    size="small"
    style={
      Object {
        "marginRight": "8px",
        "textTransform": "capitalize",
      }
    }
    type="primary"
  >
    London
  </Button>
  <Button
    block={false}
    ghost={false}
    key="Prague"
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    size="small"
    style={
      Object {
        "marginRight": "8px",
        "textTransform": "capitalize",
      }
    }
    type="default"
  >
    Prague
  </Button>
</div>
`;
